crypto/src/main/java/org/hyperledger/besu/crypto/AbstractSECP256.java

```diff
@@ -82,7 +76,7 @@ protected AbstractSECP256(final String curveName, final BigInteger prime) {
     curveOrder = curve.getN();
     halfCurveOrder = curveOrder.shiftRight(1);
     try {
-      keyPairGenerator = KeyPairGenerator.getInstance(ALGORITHM, PROVIDER);
+      keyPairGenerator = new ECDSA();
```

Comment:
Is this change necessary?

Reply:
Yes.  JCA and Bouncy Castle are very difficult to get working with GraalVM.  There is a lot of introspection involved and a lot of pointless overhead to get to what this change does.  We have it configured in the parameters (per ALGORITHM and PROVIDER) so that the referenced class is the only thing that can be returned.  Were this a system where we change crypto configs on the fly or if ethereum accepted multiple key lengths or algos then it would be worth it.  This is just overhead that makes AOT compilation impossible.

Reply:
I'm so sorry, I'm thoroughly confused because I assumed that it was possible to configure the curve used. I'll take a deeper look.

Reply:
The curve kind (ECDSA vs Edwards) is the same across all current configurations.  The curve parameters (secp256k1 or secp256r1) are applied to the key generator after this construction.

Also, this is not for the whole of the public key encryption used in the system, just the key pair generators.  I updated the use of imports to clarify that.